refactor: keyboards events

lemarier opened the pull request:
<!--
Please make sure to read the Pull Request Guidelines:
https://github.com/tauri-apps/tauri/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines
-->

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)
<!--
If yes, please describe the impact and migration path for existing applications in an attached issue. Filing a PR with breaking changes that has not been discussed and approved by the maintainers in an issue will be immediately closed.
-->

- [x] Yes. Issue #___
- [ ] No


**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
Fix https://github.com/tauri-apps/tao/issues/33

lemarier committed:
WIP windows keyboards

imported from winit

lemarier committed:
wip macos keyboards events

lemarier committed:
fix linux build

lemarier committed:
fix windows

lemarier committed:
make clippy happy

lemarier committed:
WIP keyboards `gtk` Implementations

lemarier committed:
wip linux

lemarier committed:
add new deps for linux

lemarier committed:
implement linux scancode

lemarier committed:
Linux modifiers

lemarier committed:
Merge remote-tracking branch 'origin/dev' into refactor/keyboard

lemarier committed:
finalize linux implementations

lemarier committed:
fmt

lemarier committed:
make clippy happy for macos

lemarier committed:
Fix `iOS` and WIP `android`

lemarier committed:
fix doc

lemarier committed:
fix android `TryInto`

lemarier committed:
make clippy happy and rename keycodes to keyboard

lemarier committed:
wip hotkey implementation

lemarier committed:
WIP implement accelerators for windows

lemarier committed:
change menuid to `u16`

lemarier committed:
fix ios & android

lemarier committed:
clean`custom_menu` example

lemarier committed:
remove lifetime specifier and fix linux

lemarier committed:
fix all examples

lemarier committed:
fix lifetime specifiers

lemarier committed:
cleanup

lemarier committed:
fix windows

lemarier committed:
make clippy happyyy

lemarier committed:
fix `MenuItemAttributes` and fmt

lemarier committed:
windows cleanup

lemarier committed:
fix `u32`

lemarier committed:
remove debug

lemarier committed:
update doc

lemarier committed:
update `hotkey` example

lemarier committed:
add current changelog

lemarier committed:
fmt

lemarier committed:
basic `global_accelerator` implementation for macOS

lemarier committed:
WIP `HotKeyManager`

lemarier committed:
WIP add `GlobalHotKeyEvent`

lemarier committed:
wip windows

lemarier committed:
fix windows build

lemarier committed:
add global events for windows

lemarier committed:
wip `gtk` global accelerator

lemarier committed:
fix linux build

lemarier committed:
wip(macos): refactor `Accelerator` and `ShortcutManager`

lemarier committed:
wip(windows): refactor `Accelerator` and `ShortcutManager`

lemarier committed:
Merge remote-tracking branch 'origin/dev' into refactor/keyboard

# Conflicts:
#	examples/custom_menu.rs
#	examples/multiwindow.rs
#	src/platform_impl/linux/event_loop.rs
#	src/platform_impl/linux/menu.rs
#	src/platform_impl/linux/window.rs
#	src/platform_impl/macos/menu.rs

lemarier committed:
make clippy happy

lemarier committed:
wip(linux): refactor `Accelerator` and `ShortcutManager`

lemarier committed:
fix linux `Accelerator`

lemarier committed:
review unregister

lemarier committed:
fix windows

lemarier committed:
wip linux global accelerator

lemarier committed:
fmt

lemarier committed:
fix `iOS` and `android`

lemarier committed:
fmt

lemarier committed:
remove accelerator limitation

lemarier committed:
better example

lemarier committed:
add `from_str` to Accelerator to be compatible with current tauri use

lemarier committed:
fix linux build

lemarier committed:
implement `AcceleratorId` and use `KeyCode` instead of `Key` for accelerator

lemarier committed:
refactor linux accelerators

lemarier committed:
migrate windows to `KeyCode` and fix CI

lemarier committed:
make clippy happy

lemarier committed:
fix warning’s

lemarier committed:
fix windows `reset_dead_keys`

lemarier committed:
fmt

lemarier committed:
make clippy happy

lemarier committed:
expose `with_id` for `Accelerator`

lemarier committed:
fix macOS default accelerator keys

lemarier committed:
[ci skip] update readme

lemarier committed:
WIP: use &str for `Key::Character`

lemarier committed:
fix windows, linux & android

lemarier committed:
update changelog

lemarier committed:
fix `text_with_all_modifiers`

lemarier committed:
add static lifetime

lemarier committed:
prevent windows to failed

lemarier committed:
add `GlobalShortcut` documentation

lemarier committed:
[ci skip] update changelog

lemarier committed:
fix `X` char

lemarier committed:
add accelerators tests

lemarier committed:
remove `to_keycode` as its not needed anymore

wusyong commented on src/platform/scancode.rs:
```diff
@@ -0,0 +1,31 @@
+))]
+
+// TODO: Maybe merge this with `modifier_supplement` if the two are indeed supported on the same
+// set of platforms
```
Yeah, I think it would be a lot better to merge them. Maybe also `global_shortcut`.
Does an integrated module call `shortcut` or `accelerator` make sense?

lemarier replied:
I'm not sure I understand this right; `Does an integrated module call shortcut or accelerator make sense?`

I've merged `to_scancode` and `from_scancode`.

lemarier committed:
move `to_scancode` and `from_scancode` builtin KeyCode

lemarier committed:
fix mobiles and fmt

lemarier committed:
Add doc

lemarier committed:
fix ci

lemarier committed:
fix windows

lemarier committed:
Merge remote-tracking branch 'origin/dev' into refactor/keyboard

lemarier committed:
remove `modifier_supplement` and merge it into `KeyEvent`

as we’ve been able to build the `text_with_all_modifiers` for all 3 majors platforms, i don’t think it worth exposing it behind a PlatformExt.

lemarier committed:
make clippy happy

lemarier committed:
fix windows typo in `text_with_all_modifiers`

lemarier committed:
remove unused imports in windows

lemarier committed:
Merge remote-tracking branch 'origin/dev' into refactor/keyboard

# Conflicts:
#	examples/custom_menu.rs
#	src/platform_impl/linux/mod.rs
#	src/platform_impl/macos/mod.rs

lemarier committed:
remove `nameof` and add `Debug` to `NativeKeyCode`

lemarier committed:
fmt

lemarier committed:
[ci skip] remove `modifier_supplement` from doc

lemarier committed:
[ci skip] fix ios and android clippy warning

lemarier committed:
fix linux clippy warning
